From 750ae7b4ff5f6e2867225920f821a64f6fc394f3 Mon Sep 17 00:00:00 2001
From: Soeren Grunewald <soeren.grunewald@avionic-design.de>
Date: Wed, 28 Mar 2012 10:00:07 +0200
Subject: [PATCH 2/3] Fix build without javascript debugger.

This will wrap the code used by javascript debugger with the required define
so it is only used when build with javascript debugger support enabled.

Signed-off-by: Soeren Grunewald <soeren.grunewald@avionic-design.de>
---
 Source/WebCore/inspector/InspectorMemoryAgent.cpp |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/Source/WebCore/inspector/InspectorMemoryAgent.cpp b/Source/WebCore/inspector/InspectorMemoryAgent.cpp
index 7242d65..74b8c29 100644
--- a/Source/WebCore/inspector/InspectorMemoryAgent.cpp
+++ b/Source/WebCore/inspector/InspectorMemoryAgent.cpp
@@ -292,8 +292,9 @@ InspectorMemoryAgent::~InspectorMemoryAgent()
 void InspectorMemoryAgent::getDOMNodeCount(ErrorString*, RefPtr<InspectorArray>& domGroups, RefPtr<InspectorObject>& strings)
 {
     CounterVisitor counterVisitor(m_page);
+#if ENABLE(JAVASCRIPT_DEBUGGER)
     ScriptProfiler::visitJSDOMWrappers(&counterVisitor);
-
+#endif
     // Make sure all documents reachable from the main frame are accounted.
     for (Frame* frame = m_page->mainFrame(); frame; frame = frame->tree()->traverseNext()) {
         if (Document* doc = frame->document())
diff --git a/Source/WebKit/gtk/webkit/webkitwebinspector.cpp b/Source/WebKit/gtk/webkit/webkitwebinspector.cpp
index 692079f..34b33a5 100644
--- a/Source/WebKit/gtk/webkit/webkitwebinspector.cpp
+++ b/Source/WebKit/gtk/webkit/webkitwebinspector.cpp
@@ -346,14 +346,13 @@ static void webkit_web_inspector_finalize(GObject* object)
 
 static void webkit_web_inspector_set_property(GObject* object, guint prop_id, const GValue* value, GParamSpec* pspec)
 {
-    WebKitWebInspector* web_inspector = WEBKIT_WEB_INSPECTOR(object);
-    WebKitWebInspectorPrivate* priv = web_inspector->priv;
-
     switch(prop_id) {
     case PROP_JAVASCRIPT_PROFILING_ENABLED: {
 #if ENABLE(JAVASCRIPT_DEBUGGER)
-        bool enabled = g_value_get_boolean(value);
+        WebKitWebInspector* web_inspector = WEBKIT_WEB_INSPECTOR(object);
+        WebKitWebInspectorPrivate* priv = web_inspector->priv;
         WebCore::InspectorController* controller = priv->page->inspectorController();
+        bool enabled = g_value_get_boolean(value);
         if (enabled)
             controller->enableProfiler();
         else
-- 
1.7.9.3

